snapcraft/internal/lifecycle/_clean.py

```diff
@@ -36,56 +36,51 @@ def _reverse_dependency_tree(config, part_name):
     return dependents
 
 
-def _clean_part_and_all_dependents(part_name, step, config, staged_state,
-                                   primed_state):
-    # Obtain the reverse dependency tree for this part. Make sure all
-    # dependents are cleaned.
-    dependents = _reverse_dependency_tree(config, part_name)
-    dependent_parts = {p for p in config.all_parts
-                       if p.name in dependents}
-    for dependent_part in dependent_parts:
-        dependent_part.clean(staged_state, primed_state, step)
+def _clean_part(part_name, step, config, staged_state, primed_state):
+    if step == common.COMMAND_ORDER[-1]:
+        template = 'Cleaning {step} step for {part}'
+    else:
+        template = 'Cleaning {step} step (and all subsequent steps) for {part}'
 
-    # Finally, clean the part in question
+    logger.info(template.format(step=step, part=part_name))
     config.parts.clean_part(part_name, staged_state, primed_state, step)
 
+    # If we just cleaned the root of a dependency tree, we need to mark all
+    # its dependents as dirty so they require cleaning
```

Comment:
this is overly pedantic :roll_eyes: , sorry, but sentences here should end with a period (like you did in other places)

Reply:
Fixed.